frontend/src/metabase/visualizations/components/settings/ChartSettingInputNumeric.tsx

```diff
@@ -20,15 +20,22 @@ const ALLOWED_CHARS = [
   "e",
 ];
 
+// Note: there are more props than these that are provided by the viz settings
+// code, we just don't have types for them here.
 interface ChartSettingInputProps {
   value: number | undefined;
   onChange: (value: number | undefined) => void;
   onChangeSettings: () => void;
+  options?: {
```

Comment:
I think it would be a bit more conventional to have `isInteger` and `isNonNegative` as props

Reply:
I initially tried that, but I think for settings widgets you can't add non-standard props via the setting definition, they have to go in `options`. Standard props that are shared between all widgets like `placeholder` do work outside of `options` though.